pandas/core/internals/blocks.py

```diff
@@ -2859,16 +2854,15 @@ def _extend_blocks(result, blocks=None):
     return blocks
 
 
-def _block_shape(values, ndim=1, shape=None):
+def _block_shape(values: ArrayLike, ndim: int = 1) -> ArrayLike:
     """ guarantee the shape of the values to be at least 1 d """
     if values.ndim < ndim:
-        if shape is None:
-            shape = values.shape
-        if not is_extension_array_dtype(values):
-            # TODO: https://github.com/pandas-dev/pandas/issues/23023
+        shape = values.shape
+        if not is_extension_array_dtype(values.dtype):
+            # TODO(EA2D): https://github.com/pandas-dev/pandas/issues/23023
             # block.shape is incorrect for "2D" ExtensionArrays
             # We can't, and don't need to, reshape.
-            values = values.reshape(tuple((1,) + shape))
+            values = values.reshape(tuple((1,) + shape))  # type: ignore
```

Comment:
I _think_ OK to use cast after using is_* to avoid the #type: ignore

Reply:
https://pandas.pydata.org/docs/development/contributing.html#style-guidelines specifically mentions the is_* case and suggests a refactor is the preferred route.

> i went with the latter bc its easier to grep for and out on a list of Things To Fix

for me, this is OK. but I believe Jeff and Will oppose type:ignore as a reason to `revist later`. In the past, i've been in the minority and struggled to get PRs accepted. If you can see the benefits of a fix later strategy with type hints, now could be a good time to resurface this discussion.

pyre accepts this is a reasonable need for gradual typing, see https://pyre-check.org/docs/error-suppression.html#suppression-comment-types. it has `pyre-fixme` and `pyre-ignore`.

Reply:
@jbrockmendel #32785 was opened for this discussion. see https://github.com/pandas-dev/pandas/pull/32730#discussion_r393485655 cc @jorisvandenbossche